self_hosted/ir.oomph

```diff
@@ -42,10 +38,18 @@ export class BasicType(
         else:
             return (self.generic_source == other.generic_source)
 
+    meth hash() -> Int:
+        switch self.generic_source:
+            case null _:
+                # can rely on name
+                return self.name.hash()
+            case GenericSource s:
+                return s.generik.hash()
```

Comment:
Maybe this can always rely on `self.name`, because only unions can change name weirdly?

Reply:
name can contain item type. e.g. `List[SomeUnionName]` vs `List[Foo | Bar]` can be same